### ISSUE TITLE
Configure the catalog controller for searching, faceting, and other base config

### ISSUE THREAD

**Issue by straleyb** [OregonDigital/OD2#28]
### Descriptive summary

This ticket expressed the need of configuration of the catalog controller. This is a large request as there are many configurations that need to be set along with quite a bit of testing to make sure these configurations are set properly. A lot of configuration is already expressed here, and can be translated over based on metadata needs expressed in the MAP. >>>>>>>>>>>>> LINK TO MAP HERE >>>>>>>>>>>>> 

## Facet Field Configuration:

Facets have to be configured from the catalog controller. https://github.com/osulp/Scholars-Archive/blob/master/app/controllers/catalog_controller.rb#L72 This means translating the MAP information over into the catalog controller based on what is faceted or not. Some of the fields may be faceted, but are also ordered, or controlled and more work may need to be done with those. That information will be expressed in another ticket however and for starters, getting the basic metadata fields that need to be faceted set up first. 

```
    config.add_facet_field solr_name('contributor_committeemember', :facetable), limit: 5, label: 'Committee Member'

```

This configuration takes the name of a field and the facetable symbol, a limit to how many values show up on the initial view of the facet, and the label to be displayed on the facet tab on the search page. 

<img width="1360" alt="screen shot 2018-10-04 at 9 33 15 am" src="https://user-images.githubusercontent.com/6424683/46488799-94d99f00-c7b8-11e8-8c80-ce3fa0ce9031.png">

## Index Field Configuration(Search result metadata displayed per search result): 

The Index Fields are the fields that appear in the search result list for each work. https://github.com/osulp/Scholars-Archive/blob/master/app/controllers/catalog_controller.rb#L122
For instance, title, creator, subject, may be pieces of metadata that we need to show up per item in the search results. This information should be expressed in the map and will need to be configured here.

```
config.add_index_field solr_name("title", :stored_searchable), label: "Title", itemprop: 'name', if: false
```

This configuration takes the field name, and stored searchable as the symbol. The label of the search field, the itemprop, and sometimes a helper method. The helper method is used to process a certain chain of methods to change or render specific fields depending on the results. 

<img width="912" alt="screen shot 2018-10-04 at 9 37 41 am" src="https://user-images.githubusercontent.com/6424683/46489039-395be100-c7b9-11e8-9dcd-7415c21fd0be.png">

## Show field Configuration

The show page or the single item page will need to be configured here https://github.com/osulp/Scholars-Archive/blob/master/app/controllers/catalog_controller.rb#L150
This configures the show page to display the proper information within the system. There may be some extra customization that we need to do to show proper information per work type, but initially this is a good place to start to get the right data showing up on the show page.

```
config.add_show_field solr_name("publisher", :stored_searchable), label: "Publisher"
```

This configuration takes the field name and the stored searchable symbol and a label.

This configuration is the same every time and replace the label and solr name with the proper label and field name.

<img width="1351" alt="screen shot 2018-10-04 at 9 39 03 am" src="https://user-images.githubusercontent.com/6424683/46489088-5f818100-c7b9-11e8-852a-2b3e873782df.png">

## Search Fields

The search field configuration is done in two parts. The first part is the "all fields" search which is a simple search in the search bar. The main spot to pay attention to is here: https://github.com/osulp/Scholars-Archive/blob/master/app/controllers/catalog_controller.rb#L200 This is a list of the fields that will be searched when someone enters a search in the search bar. Each field that is required for searching will have to be input here.

Second, there is https://github.com/osulp/Scholars-Archive/blob/master/app/controllers/catalog_controller.rb#L244-L251 this configuration. This allows the field to actually be searched if it is a faceted search. The facet may show up if it has been configured as a facet, but the system wont know how to search for that specific field unless it is also configured as a search field here.

### Expected behavior

The show page, searching, facets, search result metadata all should appear correctly throughout the system

### Related work

#27

**Comment by vantuyls** [comment]
requires data to be in the repository.

**Comment by straleyb** [comment]
Blocked by #57

**Comment by raybrarian** [comment]
@straleyb  - looks like #57 is no longer a blocker as the seed data has been generated and made available here: [https://oregonstate.box.com/s/jw4767z4w1sntsch35r0wjvnah2ptldy]. Does that resolve the block?

There are some caveats in the issue about downloading - "Box has a 15GB limit for all accounts, so that won't work. You can select multiple files/folders still, as long as the total doesn't go over 15GB."

**Comment by jsimic** [comment]
POSM reviewed. Closed.
